### DISCUSSION TITLE
Don't show recommendations below a certain threshold during import

### DISCUSSION THREAD

**Discussion by arsaboo** [beetbox/beets#4799] [Q&A]
Is it possible to ignore recommendations below a certain threshold (say 60%) during the import? That is, only show recommendations that match at least 60%. I want to do a quick check on my library to see if there are any new matches on MB. SO, I would like to use the `--timid` option but ignore all low-quality matches. 

Basically, I want to skip over if there are no good (over 60%) matches and confirm with me if there are good matches. 

I played with `medium_rec_thresh` but that does not seem to exclude low matches.

**Comment by sampsyo** [comment]
Hmm; I don't think we have a way to ignore these matches altogether. But you're right to play with `medium_rec_thresh`, which will at least avoid showing you a "best match" to confirm when all the matches are low-quality and instead show you the list to pick from.

It sounds like what you'd want is to *automatically skip* when all the matches are bad. This is not quite that, but you could imagine setting `default_action` to skip to at least make it easy to skip stuff (just hit the return key):
https://beets.readthedocs.io/en/stable/reference/config.html#default-action

**Reply by arsaboo** [reply]
It would be nice to have a config setting to ignore low-quality matches.

**Reply by arsaboo** [reply]
@sampsyo I am trying to add this feature, but I am a little unsure how `distance` gets added to an album. I see that plugins have `get_albums` (e.g., [here](https://github.com/beetbox/beets/blob/430a75d1ad0347913497cdb24f1e68ae146c5248/beetsplug/discogs.py#L269))function that return a list of potential albums. However, it is not clear to me how can I get the album distance for each of these albums. I see that plugins have a `album_distance` function but it seems it is applied automatically. Is it possible to get the distance value for each album? Does `get_albums` automatically return sorted by distance?

**Reply by sampsyo** [reply]
Here's where distances come from:
https://github.com/beetbox/beets/blob/5aae612f45716cfdea2884e3ae873c58e59ef64e/beets/autotag/match.py#L421-L434

In particular, in `_add_candidate`, where we ingest a new candidate from MB or a plugin, we call the `distance` function to calculate its quality and the construct an `AlbumMatch` object that includes it.
